crates/types/src/vote.rs

```diff
@@ -255,7 +319,448 @@ pub trait Accumulator<T, U>: Sized {
+    /// A quorum's worth of stake, generally 2f + 1
+    pub success_threshold: NonZeroU64,
+    /// A list of valid signatures for certificate aggregation
+    pub sig_lists: Vec<<BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature>,
+    /// A bitvec to indicate which node is active and send out a valid signature for certificate aggregation, this automatically do uniqueness check
+    pub signers: BitVec,
+    /// Phantom data to specify the vote this accumulator is for
+    pub phantom: PhantomData<VOTE>,
+}
+
+impl<
+        TYPES: NodeType,
+        COMMITTABLE: Committable + Serialize + Clone,
+        VOTE: VoteType<TYPES, COMMITTABLE>,
+    > Accumulator2<TYPES, COMMITTABLE, VOTE> for DAVoteAccumulator<TYPES, COMMITTABLE, VOTE>
+{
+    fn append(
+        mut self,
+        vote: VOTE,
+        vote_node_id: usize,
+        stake_table_entries: Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>,
+    ) -> Either<Self, AssembledSignature<TYPES>> {
+        let VoteData::DA(vote_commitment) = vote.get_data() else {
+            return Either::Left(self);
+        };
+
+        let encoded_key = vote.get_key().to_bytes();
+
+        // Deserialize the signature so that it can be assembeld into a QC
+        // TODO ED Update this once we've gotten rid of EncodedSignature
+        let original_signature: <BLSOverBN254CurveSignatureScheme as SignatureScheme>::Signature =
+            bincode_opts()
+                .deserialize(&vote.get_signature().0)
+                .expect("Deserialization on the signature shouldn't be able to fail.");
+
+        let (da_stake_casted, da_vote_map) = self
+            .da_vote_outcomes
+            .entry(vote_commitment)
+            .or_insert_with(|| (0, BTreeMap::new()));
+
+        // Check for duplicate vote
+        // TODO ED Re-encoding signature key to bytes until we get rid of EncodedKey
+        // Have to do this because SignatureKey is not hashable
+        if da_vote_map.contains_key(&encoded_key) {
+            return Either::Left(self);
+        }
+
+        if self.signers.get(vote_node_id).as_deref() == Some(&true) {
+            error!("Node id is already in signers list");
+            return Either::Left(self);
+        }
+        self.signers.set(vote_node_id, true);
+        self.sig_lists.push(original_signature);
+
+        // Already checked that vote data was for a DA vote above
+        *da_stake_casted += u64::from(vote.get_vote_token().vote_count());
+        da_vote_map.insert(
+            encoded_key,
+            (vote.get_signature(), vote.get_data(), vote.get_vote_token()),
+        );
+
+        if *da_stake_casted >= u64::from(self.success_threshold) {
+            // Assemble QC
+            let real_qc_pp = <TYPES::SignatureKey as SignatureKey>::get_public_parameter(
+                // TODO ED Something about stake table entries.  Might be easier to just pass in membership?
+                stake_table_entries.clone(),
```

Comment:
This is relevant to what parameters are needed when [assembling](https://github.com/EspressoSystems/HotShot/blob/main/crates/hotshot-qc/src/bit_vector_old.rs#L71) signatures. If we pass in membership here, then we'll need to extract these stake table entries somewhere deeper, but it's true that our input format of the append val is a little bit messy :( Not sure whether there's a better way to relay the information into append().

Reply:
But this val format is already much better than before! I guess it's fine.